go/vumitools/app_worker.py

```
@@ -414,3 +429,112 @@ def _publish_message(self, message, endpoint_name=None):
             message, endpoint_name)
+
+
+class GoRouterConfig(BaseWorker.CONFIG_CLASS, GoWorkerConfigMixin):
+    ri_connector_name = ConfigText(
+        "The name of the receive_inbound connector.",
+        required=True, static=True)
+    ro_connector_name = ConfigText(
+        "The name of the receive_outbound connector.",
+        required=True, static=True)
+
+
+class GoRouterWorker(GoRouterMixin, BaseWorker):
+    """
+    A base class for Vumi Go router worker.
+
+    Configuration parameters:
+
+    :type worker_name: str
+    :param worker_name:
+        The name of this worker, used for receiving control messages.
+
+    """
+    CONFIG_CLASS = GoRouterConfig
+
+    worker_name = None
+
+    def setup_router(self):
+        return self._go_setup_worker()
+
+    def teardown_router(self):
+        return self._go_teardown_worker()
+
+    def _publish_message(self, message, endpoint_name=None):
+        if not self.get_metadata_helper(message).get_conversation_info:
+            log.error(
+                "Conversation metadata missing for message for %s: %s" % (
+                    type(self).__name__, message))
+        return super(GoApplicationWorker, self)._publish_message(
+            message, endpoint_name)
+
+    def setup_worker(self):
+        d = maybeDeferred(self.setup_router)
+        d.addCallback(lambda r: self.unpause_connectors())
+        return d
+
+    def teardown_worker(self):
+        self.pause_connectors()
+        return self.teardown_router()
+
+    def handle_inbound(self, config, msg):
+        raise NotImplementedError()
+
+    def handle_outbound(self, config, msg):
+        raise NotImplementedError()
+
+    def handle_event(self, config, event):
+        raise NotImplementedError()
+
+    def _mkhandler(self, handler_func, connector_name):
+        def errback(f):
+            log.error("Error routing message for %s" % (connector_name,))
+            log.error(f)
```

This swallows error messages which will be an issue if higher level stuff wants to publish `nacks` ever. Maybe not an issue for now though?